ios/browser/api/ui/webui/skus/skus_internals_ui.mm

```diff
@@ -0,0 +1,285 @@
+      },
+      browser_state);
+
+  // Bind Mojom Interface
+  web_ui->GetWebState()->GetInterfaceBinderForMainFrame()->AddInterface(
+      base::BindRepeating(&SkusInternalsUI::BindInterface,
+                          base::Unretained(this)));
+}
+
+SkusInternalsUI::~SkusInternalsUI() {}
+
+void SkusInternalsUI::BindInterface(
+    mojo::PendingReceiver<skus::mojom::SkusInternals> pending_receiver) {
+  if (skus_internals_receiver_.is_bound()) {
+    skus_internals_receiver_.reset();
+  }
+
+  skus_internals_receiver_.Bind(std::move(pending_receiver));
+}
+
+void SkusInternalsUI::GetEventLog(GetEventLogCallback callback) {
+  // TODO(simonhong): Ask log to SkusService
+  NOTIMPLEMENTED_LOG_ONCE();
+}
+
+void SkusInternalsUI::GetSkusState(GetSkusStateCallback callback) {
+  std::move(callback).Run(GetSkusStateAsString());
+}
+
+void SkusInternalsUI::GetVpnState(GetVpnStateCallback callback) {
+  base::Value::Dict dict;
+  dict.Set("Order", GetOrderInfo("vpn."));
+  std::string result;
+  base::JSONWriter::Write(dict, &result);
+  std::move(callback).Run(result);
+}
+
+void SkusInternalsUI::GetLeoState(GetLeoStateCallback callback) {
+  base::Value::Dict dict;
+  dict.Set("Order", GetOrderInfo("leo."));
+  std::string result;
+  base::JSONWriter::Write(dict, &result);
+  std::move(callback).Run(result);
+}
+
+base::Value::Dict SkusInternalsUI::GetOrderInfo(
+    const std::string& location) const {
+  base::Value::Dict dict;
+
+  const auto& skus_state = local_state_->GetDict(skus::prefs::kSkusState);
+  for (const auto kv : skus_state) {
+    if (!base::StartsWith(kv.first, "skus:")) {
+      continue;
+    }
+
+    // Convert to Value as it's stored as string in local state.
+    auto json_value = base::JSONReader::Read(kv.second.GetString());
+    if (!json_value) {
+      continue;
+    }
+
+    const auto* skus = json_value->GetIfDict();
+    if (!skus) {
+      continue;
+    }
+
+    const auto* orders = skus->FindDict("orders");
+    if (!orders) {
+      continue;
+    }
+
+    base::Value::Dict order_dict_output;
+    for (const auto order : *orders) {
+      const auto* order_dict = order.second.GetIfDict();
+      if (!order_dict) {
+        continue;
+      }
+
+      if (auto* order_location = order_dict->FindString("location")) {
+        if (!base::StartsWith(*order_location, location)) {
+          continue;
+        }
+        order_dict_output.Set("location", *order_location);
+      }
+
+      if (auto* id = order_dict->FindString("id")) {
+        order_dict_output.Set("id", *id);
+      }
+      if (auto* expires_at = order_dict->FindString("expires_at")) {
+        order_dict_output.Set("expires_at", *expires_at);
+      }
+    }
+
+    // Set output with env like {skus:production: {...}}.
+    dict.Set(kv.first, std::move(order_dict_output));
+  }
+  return dict;
+}
+
+void SkusInternalsUI::ResetSkusState() {
+  // TODO: Ask iOS side to reset skus
```

Comment:
Can we not just `<if expr="not is_ios">` the button in the actual WebUI html instead? Better to just not show a button that isn't working on iOS

Reply:
No longer needed. It works fine since I fixed the alerts not working.